Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -255,53 +264,56 @@ class DatabaseContainer: NSPersistentContainer {
     /// Tries to load a persistent store.
     ///
     /// If it fails, for example because of non-matching models, it removes the store, recreates is, and tries to load it again.
-    /// If the second loading fails, too, it throws an error.
     ///
-    private func setupPersistentStore() throws {
-        var storeLoadingError: Error?
-        
-        loadPersistentStores { _, error in
-            storeLoadingError = error
-        }
-        
-        if storeLoadingError != nil {
-            try recreatePersistentStore()
+    private func setupPersistentStore() {
```

Comment:
Can we also add a completion closure here? This one is called from `DatabaseContainer.init` and followed by `resetEphemeralValues` which can also bite us back if we don't wait for the store to be loaded/recreated.